src/main/java/mate/academy/spring/dao/AuthorDaoImp.java

```diff
@@ -0,0 +1,49 @@
+package mate.academy.spring.dao;
+
+import java.util.List;
+
+import javax.persistence.TypedQuery;
+
+import mate.academy.spring.dao.interfaces.AuthorDao;
+import mate.academy.spring.entity.Author;
+import org.hibernate.SessionFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Repository;
+
+@Repository
+public class AuthorDaoImp implements AuthorDao {
+
+    @Autowired
+    private SessionFactory sessionFactory;
+
+    @Override
+    public void add(Author author) {
+        sessionFactory.getCurrentSession().save(author);
+    }
+
+    @Override
+    public List<Author> listAuthor() {
+        @SuppressWarnings("unchecked")
+        TypedQuery<Author> query = sessionFactory.getCurrentSession().createQuery("from Author");
+        return query.getResultList();
+    }
+
+    @Override
+    public List<Author> findByName(String name) {
+        @SuppressWarnings("unchecked")
```

Comment:
@kenu21 для того щоб не писати це, раджу використати щось типу такого:

```suggestion
                    Query<Author> query = sessionFactory.getCurrentSession()
                    .createQuery("FROM Author WHERE name LIKE concat('%',:name,'%') "
                            + "ORDER BY id DESC", Author.class);
```